### PR TITLE
Test azure again

### PR DESCRIPTION
**Pull request by trieloff**
See #31

### CODE REVIEW

**Comment by trieloff** [comment]
@stefan-guggisberg I'd be glad for some help here.

The `updatePackage()` in `AzureDeployer` should POST a JSON to `https://deploy-helix.scm.azurewebsites.net/api/settings` but the response is consistent with the request body being empty.

With verbose logging enabled, I see this:

```
--: updating package on Azure ...
--: updating app (package) parameters ...
STREAM 3899: read 0
STREAM 3899: need readable false
STREAM 3899: length less than watermark true
STREAM 3899: do read
STREAM 3899: read 0
STREAM 3899: need readable true
STREAM 3899: length less than watermark true
STREAM 3899: reading or ended false
  helix-fetch:core established TLS connection: #1 (deploy-helix.scm.azurewebsites.net) +0ms
  helix-fetch:core deploy-helix.scm.azurewebsites.net -> http/1.1 +0ms
  helix-fetch:h1 POST https://deploy-helix.scm.azurewebsites.net/api/settings +0ms
  helix-fetch:h1 reusing socket #1 (deploy-helix.scm.azurewebsites.net) +0ms
STREAM 3899: pipe count=1 opts=undefined
STREAM 3899: resume
STREAM 3899: resume
STREAM 3899: resume false
STREAM 3899: read 0
STREAM 3899: need readable false
STREAM 3899: length less than watermark true
STREAM 3899: do read
STREAM 3899: flow true
STREAM 3899: read undefined
STREAM 3899: need readable true
STREAM 3899: length less than watermark true
STREAM 3899: reading or ended false
STREAM 3899: resume true
STREAM 3899: flow true
STREAM 3899: read undefined
STREAM 3899: need readable true
STREAM 3899: length less than watermark true
STREAM 3899: reading or ended false
STREAM 3899: readableAddChunk {
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk "
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk H
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk E
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk Y
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk "
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk :
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk "
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk h
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk o
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk "
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk }
STREAM 3899: ondata
HTTP 3899: write ret = true
STREAM 3899: dest.write true
STREAM 3899: readableAddChunk null
STREAM 3899: onEofChunk
STREAM 3899: emitReadable_ false 0 true
STREAM 3899: flow true
STREAM 3899: read undefined
STREAM 3899: endReadable false
STREAM 3899: endReadableNT false 0
STREAM 3899: onend
HTTP 3899: outgoing message end.
STREAM 3899: onWriteComplete 0 undefined
STREAM 3899: onWriteComplete 0 undefined
STREAM 3899: onfinish
STREAM 3899: unpipe
STREAM 3899: onunpipe
STREAM 3899: cleanup
STREAM 3899: readableAddChunk <Buffer 48 54 54 50 2f 31 2e 31 20 34 30 30 20 42 61 64 20 52 65 71 75 65 73 74 0d 0a 43 61 63 68 65 2d 43 6f 6e 74 72 6f 6c 3a 20 6e 6f 2d 63 61 63 68 65 0d ... 576 more bytes>
HTTP 3899: AGENT incoming response!
STREAM 3899: readableAddChunk null
STREAM 3899: onEofChunk
STREAM 3899: emitReadable false false
STREAM 3899: emitReadable null
STREAM 3899: emitReadable_ false 0 true
STREAM 3899: flow null
STREAM 3899: maybeReadMore read 0
STREAM 3899: read 0
STREAM 3899: need readable true
STREAM 3899: length less than watermark true
STREAM 3899: do read
```

so the request body (`{"HEY":"ho"}`) seems to be there, but still somehow missing.

**Comment by trieloff** [comment]
More (mystifying) data:

Swapping the Azure endpoint with a Pipedream URL, I can see that the request arrives as expected, including the POST body.

**Comment by trieloff** [comment]
Replacing `helix-fetch` with `node-fetch` works, filing this as a `helix-fetch` issue.

**Comment by trieloff** [comment]
Blocked by adobe/helix-fetch#142

File: src/ActionBuilder.js [hunk 1]
```diff
@@ -310,7 +334,7 @@ module.exports = class ActionBuilder {
 
     archive.append(JSON.stringify(packageJson, null, '  '), { name: 'package.json' });
     // azure functions manifest
-    archive.file(path.resolve(__dirname, 'template', 'function.json'), { name: 'function.json' });
+    archive.append(JSON.stringify(this.functionJson, null, '  '), { name: 'function.json' });
```
Comment (tripodsan) [inline src/ActionBuilder.js:337]:
```suggestion
    archive.append(JSON.stringify(this.functionJson, null, 2), { name: 'function.json' });
```

**Review by tripodsan — APPROVED** [review]
ok, but suffix handling is still missing in `template.js`

File: src/template/index.js [hunk 2]
```diff
@@ -116,15 +116,15 @@ async function azure(context, req) {
...
+        suffix: `/${req.url.split('/').slice(7).join('/')}`,
       },
       runtime: {
         name: 'azure-functions',
         region: process.env.Location,
       },
       func: {
         name: context.executionContext.functionName,
-        version: undefined, // seems impossible to get
+        version: req.url.split('/').slice(6, 7)[0],
```
Comment (tripodsan) [inline src/template/index.js:127]:
```suggestion
        version: req.url.split('/')[6],
```

**Review by tripodsan — CHANGES_REQUESTED** [review]
missing test for `suffix`

**Review by tripodsan — APPROVED** [review]

**Pull request merged**